### PR TITLE
feat: adjust encryption-service-vault exceptions

### PR DESCRIPTION
**Pull request by mattnichols**
Update vault encryption service to use new Path SDK Exceptions

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy [hunk 1]
```diff
@@ -427,6 +565,80 @@ class VaultEncryptionServiceTest extends Specification {
...
+
+    then:
+    key == null
+    verify(subject, never()).resetDriver() || true
+  }
+
+
+
+
+
```
Comment (stevecl5) [inline encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy]: extra blank lines

File: encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy [hunk 2]
```diff
@@ -18,6 +21,7 @@ import com.mx.common.collections.ObjectMap
 
 import org.mockito.Mockito
 
+import spock.lang.Ignore
```
Comment (stevecl5) [inline encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy]: unused import

File: encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy [hunk 3]
```diff
@@ -253,11 +257,65 @@ class VaultEncryptionServiceTest extends Specification {
     configWithAppRole() | _
   }
 
+  def "decrypt rebuilds driver and re-authenticates on authentication exception"() {
+    given:
+    def config = configWithAppId()
+    subject = spy(new VaultEncryptionService(config))
+    Mockito.doReturn(vaultDriver).when(subject).buildAuthenticatedDriver(any())
```
Comment (stevecl5) [inline encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy]: `import static org.mockito.Mockito.doReturn`?

File: encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy (hunk 1 above)
Comment (mattnichols) [reply encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy]: fixed

File: encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy (hunk 2 above)
Comment (mattnichols) [reply encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy]: removed

File: encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy (hunk 3 above)
Comment (mattnichols) [reply encryption-service-vault/src/test/groovy/com/mx/path/facility/security/vault/VaultEncryptionServiceTest.groovy]: fixed

**Review by stevecl5 — APPROVED** [review]

**Pull request merged**